src/lib/web.py

```diff
@@ -0,0 +1,349 @@
+        """
+        try:
+            login_failed_header = self.driver.find_element_by_xpath(
+                "//h1[contains(text(), 'We are sorry')]",
+            )
+            if login_failed_header.is_displayed():
+                self.driver.refresh()
+        except Exception:
+            logging.debug("Login interruption did not occur !")
+            return
+
+    def open_workspace(self):
+        """
+        Opens the Eclipse CHE workspace
+        """
+        self.open_url(url=config_data["workspace_url"])
+        self.driver.maximize_window()
+        # Click on Login Dev Sandbox link
+        self.wait_click_element(by_locator_strategy=By.XPATH, locator="//a[text()='DevSandbox']")
+        self.handle_login_interrupt()
+        # Perform user login
+        self.wait_find_element(By.XPATH, "//div[@class='kc-card']")
+        login_username = self._get_workspace_username()
+        self.write_text(By.XPATH, "//input[@id='username']", login_username)
+        # Click on Next button
+        self.driver.find_element_by_xpath("//button[@id='login-show-step2']").click()
+        login_password = self._get_workspace_password()
+        self.write_text(By.XPATH, "//input[@id='password']", login_password)
+        # Click Login button
+        self.driver.find_element_by_xpath("//input[@id='kc-login']").click()
+        self.handle_login_interrupt()
+        # Wait for workspace to load and click on che-ide-tests workspace
+        self.wait_find_element(By.XPATH, "//span[@class='che-toolbar-title-label']", 60)
+        self.wait_click_element(By.XPATH, "//span[contains(text(), 'che-ide-tests')]")
+        workspace_status_ele = self.wait_find_element(By.XPATH, "//span[@id='workspace-status']")
+        workspace_status = str(workspace_status_ele.text).lower()
+        final_workspace_status = ""
+        if workspace_status == "stopped":
+            self.driver.find_element_by_xpath("//button[contains(text(), 'Run')]").click()
+        elif workspace_status == "running":
+            final_workspace_status = "running"
+        cutoff = time.time() + 150
+        while workspace_status == "stopped" or workspace_status == "starting":
+            status_ele = self.driver.find_element_by_xpath("//span[@id='workspace-status']")
+            current_status = str(status_ele.text).lower()
+            if current_status == "running":
+                final_workspace_status = current_status
+                break
+            if time.time() > cutoff:
+                break
+            time.sleep(2)
+        if final_workspace_status != "running":
+            raise Exception("The workspace could not be started !")
+        self.wait_click_element(By.XPATH, "//a[contains(text(), 'Open')]")
+        # Check workspace has loaded
+        self.wait_switch_frame(By.ID, "ide-application-iframe", 90)
+
+    def run_analysis(self):
+        """
+        Runs analysis on project in eclipse che workspace and returns story points
```

Comment:
Hi @nitishSr , I think this can be divided into two methods , create_configuration and run_analysis . 
It will help in automating test cases where we have to create configuration and then cancel.

Reply:
Will update

Reply:
@sshveta , made changes